content_script.js

```diff
@@ -0,0 +1,82 @@
+var filteredTags = ["script", "noscript", "style"];//, "link"];
+
+function serializeTree(html, holes, element, depth) {
+	if (!element.tagName && element.nodeType != 3) {
+		// ignore elements that don't have tags and are not text
+	}
+	else if (element.tagName && filteredTags.indexOf(element.tagName.toLowerCase()) != -1) {
+		// filter out elements that are in filteredTags
+	}
+	else if (element.nodeType == 3) {
+		html.push(element.textContent);
+	}
+	else {
+		html.push(new Array(depth+1).join("  "));
```

Comment:
Please add a short comment documenting what this function does and the meaning of each argument.